frontend/src/metabase/collections/components/CollectionSidebar/Bookmarks/Bookmarks.tsx

```diff
@@ -22,22 +23,40 @@ import {
 import { Bookmark, BookmarkableEntities, Bookmarks } from "metabase-types/api";
 
 interface LabelProps {
-  name: string;
-  display?: string;
-  type: BookmarkableEntities;
+  bookmark: Bookmark;
 }
 
 interface CollectionSidebarBookmarksProps {
   bookmarks: Bookmarks;
   deleteBookmark: (id: string, type: string) => void;
 }
 
-const Label = ({ display, name, type }: LabelProps) => {
-  const iconName = getIcon(display, type);
+interface IconProps {
+  name: string;
+  tooltip?: string;
+}
+
+const BookmarkIcon = ({ icon }: { icon: IconProps }) => (
+  <BookmarkTypeIcon
+    color={icon.tooltip ? color("warning") : color("brand")}
+    name={icon.name}
+    opacity={icon.tooltip ? 1 : 0.5}
+  />
+);
```

Comment:
Relying on the `tooltip` property to handle official collections looks pretty fragile. If another entity will start adding a `tooltip` to its icon prop, its icons will become yellow for an unknown reason 🤔 WDYT about something like:

```jsx
// bookmark entity

function getIcon(bookmark) {
  // using random names here, the idea is that we don't change what entities `getIcon` returned
  return getEntityForBookmark(bookmark).objectSelectors.getIcon();
}

// Bookmarks.tsx

const BookmarkIcon = ({ bookmark }) => {
   const icon = Bookmarks.objectSelectors.getIcon(bookmark)
   const isCollection = bookmark.type === "collection"
   const isRegularCollection = isCollection && PLUGIN_COLLECTIONS.isRegularCollection(bookmark)
   return (
      <BookmarkTypeIcon 
         {...icon}
         opacity={!isCollection || isRegularCollection ? 0.5 : 1}
      />
   )
}

// OR I guess we can use CollectionIcon component from
// https://github.com/metabase/metabase/blob/master/frontend/src/metabase/collections/components/CollectionIcon.jsx

const BookmarkIcon = ({ bookmark }) => {
   const isCollection = bookmark.type === "collection"
   if (isCollection) {
      return <StyledCollectionIcon collection={bookmark} />
   }
   return (
      <BookmarkTypeIcon {...icon} />
   )
}
```

Reply:
@kulyk Wdyt of us returning an `isOfficial` key in `frontend/src/metabase/entities/collections.js@getCollectionIcon`?

[In this object](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/entities/collections.js#L157) to be more precise? 

The intention is to have a single source defining when an icon would be for "official".

Reply:
This is making me think that the BE could return an `is_official` boolean in items inside the bookmarks array.

Would clean up code potentially everywhere we render a bookmark.

If we later add the possibility of "official" to entities other than a collection, it will just work.

And it would remove the brittleness and lack of clarity of checking for `!isCollection || isRegularCollection`.

Reply:
The BE already returns `authority_level: "official"` for bookmarked official collections. I believe there's no easy way to generalize this code. By "official collections" feature requirements, their icons should be rendered in their own manner in the sidebar (yellow color, opaque). So if other sidebar icons have a different default color and opacity, I believe we can't avoid the "is it an official collection" check anywhere. And I'd personally vote for a bit more ugly code, but more straightforward code

> If we later add the possibility of "official" to entities other than a collection, it will just work.

I'm pretty sure it's not going to happen. Items saved into official collections are already considered "official" right now. They're bumped in search results and they have the "official" icon shown next to their collection's title whenever it's rendered (QB header, question in dashboards, search results, data picker)

Reply:
Yeah...responsibility split between the BE and FE is peculiar in this codebase.

If only the BE didn't send `authority_level: official` for official collections if customers were in OS versions, that might be the start of a different story.

Anyway, onwards here.

Thanks.